apps/minifront/src/fetchers/registry.ts

```diff
@@ -0,0 +1,30 @@
+import { getChainId } from './chain-id';
+import { ChainRegistryClient } from '@penumbra-labs/registry';
+import { Metadata } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/asset/v1/asset_pb';
+
+export const getIbcConnections = async () => {
+  const chainId = await getChainId();
+  if (!chainId) throw new Error('Could not fetch chain id');
+
+  const registryClient = new ChainRegistryClient();
+  const { ibcConnections } = await registryClient.get(chainId);
+  return ibcConnections;
+};
+
+export const getAssetsFromRegistry = async () => {
+  const chainId = await getChainId();
+  if (!chainId) throw new Error('Could not fetch chain id');
+
+  const registryClient = new ChainRegistryClient();
+  const { assetById } = await registryClient.get(chainId);
+  return Object.values(assetById).map(m => Metadata.fromJson(m));
+};
```

Comment:
instead of using this fn to collect metadata for all assets, which will be a large amount of data, simply request asset metadata on demand from the extension when necessary. the extension should be responsible for handling the registry, and minifront probably shouldn't know about the registry at all.